src/sentry/ownership/grammar.py

```diff
@@ -109,29 +109,47 @@ def test(self, data):
         elif self.type.startswith("tags."):
             return self.test_tag(data)
         elif self.type == CODEOWNERS:
-            return self.test_codeowners(data)
+            return self.test_frames(
+                data,
+                ["filename", "abs_path"],
+                # Codeowners has a slightly different syntax compared to issue owners
+                # As such we need to match it using gitignore logic.
+                # See syntax documentation here:
+                # https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/creating-a-repository-on-github/about-code-owners
+                lambda val, pattern: _path_to_regex(pattern).search(val),
+            )
         return False
 
-    def test_url(self, data):
+    def test_url(self, data: Union[Mapping[str, Any], Sequence[Any]]) -> bool:
+        if not isinstance(data, Mapping):
+            return False
+
         try:
             url = data["request"]["url"]
         except KeyError:
             return False
-        return url and glob_match(url, self.pattern, ignorecase=True)
-
-    def test_frames(self, data, keys):
-        for frame in _iter_frames(data):
+        return url and bool(glob_match(url, self.pattern, ignorecase=True))
+
+    def test_frames(
+        self,
+        data: Union[Mapping[str, Any], Sequence[Any]],
+        keys: Sequence[str],
+        match_frame_value_func: Callable[[Any, Any], Any] = lambda val, pattern: glob_match(
```

Comment:
I think the callable would be `[[Optional[str], str], Any]`? I think it would crash with any other value, since we can't apply a regex to a non string.

The return value is a bit trickier. Essentially we only care about this in terms of returning a boolean value, although `.search` returns a match object, and `glob_match` returns a bool